src/App.tsx

```diff
@@ -14,42 +16,88 @@ import { Provider as StampsProvider } from './providers/Stamps'
-                <PlatformProvider>
-                  <TopUpProvider>
-                    <SnackbarProvider>
-                      <Router>
-                        <>
-                          <CssBaseline />
-                          <Dashboard>
-                            <BaseRouter />
-                          </Dashboard>
-                        </>
-                      </Router>
-                    </SnackbarProvider>
-                  </TopUpProvider>
-                </PlatformProvider>
-              </FeedsProvider>
-            </FileProvider>
-          </StampsProvider>
-        </BeeProvider>
-      </SettingsProvider>
-    </ThemeProvider>
-  </div>
-)
+if (config.SENTRY_KEY) {
+  Sentry.init({
+    dsn: config.SENTRY_KEY,
+    release: packageJson.version,
+    integrations: [new BrowserTracing({ tracingOrigins: ['localhost'] })],
+    tracesSampleRate: 1.0,
+    beforeSend: async (event, hint) => {
+      hint.attachments = []
+
+      try {
+        // This will fail if we are not running in Bee Desktop, but that is alright
+        hint.attachments.push({ filename: 'bee-desktop.log', data: await getBeeDesktopLogs() })
+        // eslint-disable-next-line no-empty
+      } catch (e) {}
+
+      try {
+        // This will fail if we are not running in Bee Desktop, but that is alright
+        hint.attachments.push({ filename: 'bee.log', data: await getBeeLogs() })
+        // eslint-disable-next-line no-empty
+      } catch (e) {}
+
+      return event
+    },
+  })
+}
+
+const App = ({ beeApiUrl, beeDebugApiUrl, lockedApiSettings }: Props): ReactElement => {
+  const mainApp = (
+    <div className="App">
+      <ThemeProvider theme={theme}>
+        <SettingsProvider beeApiUrl={beeApiUrl} beeDebugApiUrl={beeDebugApiUrl} lockedApiSettings={lockedApiSettings}>
+          <BeeProvider>
+            <StampsProvider>
+              <FileProvider>
+                <FeedsProvider>
+                  <PlatformProvider>
+                    <TopUpProvider>
+                      <SnackbarProvider>
+                        <Router>
+                          <>
+                            <CssBaseline />
+                            <Dashboard>
+                              <BaseRouter />
+                            </Dashboard>
+                          </>
+                        </Router>
+                      </SnackbarProvider>
+                    </TopUpProvider>
+                  </PlatformProvider>
+                </FeedsProvider>
+              </FileProvider>
+            </StampsProvider>
+          </BeeProvider>
+        </SettingsProvider>
+      </ThemeProvider>
+    </div>
+  )
+
+  // Displays Report Dialog when some component crashes
+  if (config.SENTRY_KEY) {
+    return (
+      <Sentry.ErrorBoundary
+        showDialog
+        fallback={({ error, componentStack, resetError }) => <ItsBroken message={error.message} />}
+      >
+        {mainApp}
+      </Sentry.ErrorBoundary>
+    )
+  }
```

Comment:
not that we are relying on it in CSS, but please note that this changes the component nesting. Before we had directly `<div id="app">...</div>` now this is nested in the sentry error boundary.